public/app/features/explore/ExploreGraphNGPanel.tsx

```diff
@@ -57,7 +59,26 @@ export function ExploreGraphNGPanel({
 }: Props) {
   const theme = useTheme();
   const [showAllTimeSeries, setShowAllTimeSeries] = useState(false);
-  const [structureRev, setStructureRev] = useState(1);
+  const [baseStructureRev, setBaseStructureRev] = useState(1);
+  const initialStructureRev: { previousData?: DataFrame[]; structureChanges: number } = {
+    previousData: undefined,
+    structureChanges: 0,
+  };
+  const structureRevRef = useRef(initialStructureRev);
```

Comment:
Think you can use `usePrevious`  from react-use that we use. Use 😸

Reply:
Oh nice! Didn't know we use react-use. I'll use react-use! So useful! 😃